src/Core/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -150,41 +150,92 @@ public Type GetColumnSystemType(string columnName)
         /// valid foreign key definition. It is guaranteed at least one fk definition
         /// will be valid since the MetadataProvider.ValidateAllFkHaveBeenInferred.
         /// </summary>
+        /// <param name="relationshipName">Name of the relationship as represented in config file. The GraphQL
+        /// field referencing a nested entity has the same name as the relationship.</param>
         /// <param name="targetEntityName">Entity name as in config file for the related entity.</param>
         /// <param name="relatedSourceAlias">The alias assigned for the underlying source of this related entity.</param>
         /// <param name="subQuery">The subquery to which the join predicates are to be added.</param>
-        public void AddJoinPredicatesForRelatedEntity(
+        public void AddJoinPredicatesForRelatedEntityNEW(
+            string relationshipName,
+            string targetEntityName,
+            string relatedSourceAlias,
+            BaseSqlQueryStructure subQuery)
+        {
+            EntityRelationshipKey selfJoinKey = new(EntityName, relationshipName);
```

Comment:
It's not really a selfJoin key but more like `currentEntityRelationshipKey`

Reply:
agreed.